code-studio/src/js/components/GridEditor.jsx

```diff
@@ -0,0 +1,166 @@
+/* global React, dashboard */
+
+var BeeCell = require('blockly-mooc/src/maze/beeCell');
+var Cell = require('blockly-mooc/src/maze/cell');
+var mazeUtils = require('blockly-mooc/src/maze/mazeUtils');
+var BeeCellEditor = require('./BeeCellEditor.jsx');
+var CellEditor = require('./CellEditor.jsx');
+
+window.dashboard = window.dashboard || {};
+
+window.dashboard.GridEditor = (function (React) {
+  return React.createClass({
+    propTypes: {
+      serializedMaze: React.PropTypes.arrayOf(React.PropTypes.arrayOf(React.PropTypes.object)),
+      maze: React.PropTypes.arrayOf(React.PropTypes.array), // maze items can be integers or strings
+      initialDirt: React.PropTypes.arrayOf(React.PropTypes.arrayOf(React.PropTypes.number)),
+      skin: React.PropTypes.string.isRequired,
+      onUpdate: React.PropTypes.func.isRequired
+    },
+
+    getInitialState: function () {
+      return {};
+    },
```

Comment:
You are not expected to do that. It's possible I was wrong about needing getInitialState(), but this would surprise me a little bit. If it works with it there though, I'd leave it :)